lib/ipaddr.rb

```diff
@@ -149,7 +204,7 @@ def ~
   # Returns true if two ipaddrs are equal.
   def ==(other)
     other = coerce_other(other)
-    return @family == other.family && @addr == other.to_i
+    return @family == other.family && @addr == other.to_i && mask_addr == other.mask_addr
```

Comment:
Well, that's another way of solving the spec failures. I'll just add another method to do a compare with the subnet, I'm open to suggestions for names ;)